### ISSUE TITLE
VLC Error “Your Input Can’t be Opened”

### ISSUE THREAD

**Issue by ARN1SH** [pb581/lofi-chill#1]
![image](https://user-images.githubusercontent.com/53369760/148695271-5f00f8f9-2d71-4801-b184-7d40080e4d9a.png)

Tried till 5th
https://moviemaker.minitool.com/moviemaker/vlc-your-input-cant-be-opened.html

**Comment by pb581** [comment]
That is because the MRL is expired; it's from when I uploaded the repo here (you can see that MRL expired at 1639572284 UNIX time, which was on 15 Dec 2021, around the time I uploaded the repo).

Run the `lofi_set.bat` script, not the `lofi.bat` one, to fetch a fresh playback url first, it will call other scripts as needed. Let me know if the issue is resolved.

**Comment by ARN1SH** [comment]
I saw the error after running lofi_set.bat

**Comment by pb581** [comment]
> I saw the error after running lofi_set.bat

Hmm, that is weird. That means the python script `yt_stream.pyw` that is supposed to fetch a new MRL isn't working for you. Just to make sure, have you installed the listed dependencies (`pafy`) for python for the script to work? If not, run `pip install pafy` from the command line first and then run `lofi_set.bat`.

**Comment by pb581** [comment]
I pushed a few changes to log what's happening. Clone the repo again (run `git clone https://github.com/pb581/lofi-chill/ && cd lofi-chill` if you have `git` available from command line or manually clone the repo) and then run `start lofi_set.bat`

There should be a `log.txt` generated post execution. If problem persists, post the log file here for me to see what's happening on your end. And like I said in the previous comment, you need the `pafy` module installed for the script to work.

**Comment by ARN1SH** [comment]
> > I saw the error after running lofi_set.bat
> 
> Hmm, that is weird. That means the python script `yt_stream.pyw` that is supposed to fetch a new MRL isn't working for you. Just to make sure, have you installed the listed dependencies (`pafy`) for python for the script to work? If not, run `pip install pafy` from the command line first and then run `lofi_set.bat`.


Installed the dependencies before running the script.

**Comment by ARN1SH** [comment]
It creates the log.txt but no log is outputted in it.
![image](https://user-images.githubusercontent.com/53369760/148737964-a2d13106-e7db-41c8-9079-918b6e915069.png)
However the error is outputted in the terminal.
![image](https://user-images.githubusercontent.com/53369760/148738293-d840fdef-88a9-477b-893d-cd109d3cbe66.png)

**Comment by pb581** [comment]
D'oh! I forgot to list `youtube-dl` as a required dependency.

**Comment by pb581** [comment]
I added a `requirements.txt` file to use with `pip` and an installation procedure in the README; also a hotfix for pafy that is needed (the official repo is not maintained anymore: it is a fix for https://github.com/mps-youtube/pafy/issues/306). Go to `%localappdata%\programs\python\python310\lib\site-packages\pafy\` and overwrite the `backend_youtube_dl.py` there with the one given in the `\hotfix\` directory of this repo. It should work (I checked it just now in a virtual env).
